datasetinsights/estimators/faster_rcnn.py

```diff
@@ -33,44 +36,102 @@ class FasterRCNN(Estimator):
     """
     Faster-RCNN train/evaluate implementation for object detection.
 
-    https://github.com/pytorch/vision/tree/master/references/detection
-    https://arxiv.org/abs/1506.01497
+
     Args:
         config (CfgNode): estimator config
-        writer: Tensorboard writer object
-        kfp_writer: KubeflowPipelineWriter object to write logs
-        checkpointer: Model checkpointer callback to save models
-        device: model training on device (cpu|cuda)
         box_score_thresh: (optional) default threshold is 0.05
-        gpu (int): (optional) gpu id on which code will execute
-        rank (int): (optional) rank of process executing code
+        distributed: whether or not the estimator is distributed
+        data_root: Root directory on localhost where datasets are located
+        kfp_metrics_filename: Kubeflow Metrics filename
+        kfp_metrics_dir: Path to the directory where Kubeflow
+         metrics files are stored
+
+
+    https://github.com/pytorch/vision/tree/master/references/detection
+    https://arxiv.org/abs/1506.01497
+
     Attributes:
         model: pytorch model
         writer: Tensorboard writer object
         kfp_writer: KubeflowPipelineWriter object
         checkpointer: Model checkpointer callback to save models
         device: model training on device (cpu|cuda)
+        gpu (int): (optional) gpu id on which code will execute
+        rank (int): (optional) rank of process executing code
     """
 
     def __init__(
         self,
         *,
         config,
-        writer,
```

Comment:
Why is writer, checkpointer and kfp_writer deleted here? Should they directly take input from the instances that was initialized in the `factory.create` method?

Reply:
They are deleted because they are no longer needed in the constructor as they are re-initialized later in the constructor. I dont think this is a good idea but its what we agree on during the meeting on 7 of August

Reply:
This is actually not what we discussed in the meeting. `factory.create` will initialized necessary object with defaults. We  have to do override just to "make it working for torch.distributed". So I will expect to see something like this

```
def create_estimator()
    writer = SummaryWriter(...) 
    kfp_writer = KubeflowPipelineWriter(...)
    checkpointer = EstimatorCheckpoint(...)

    return estimators_cls(
        writer=writer,
        kfp_writer=kfp_writer,
        checkpointer=checkpointer,
        **kwargs,
    )

class FasterRCNN:

def __init__(self, ..., writer, kfp_writer, checkpointer, ...):
    # note: this is a hack to make torch.distributed specific init in the class,.....
    self.writer = SummaryWriter(writer.logdir, writer_to_file=is_master())
    self.kfp_writer = kfp_writer

    checkpointer.distributed = is_distributed()
    self.checkpointer = checkpointer
```

Reply:
Will make this change